### ISSUE TITLE
[Button] Contrast for some default variants is too low

### ISSUE THREAD

**Issue by DavidJFelix** [chakra-ui/chakra-ui#1566]
**Describe the bug** According to chrome lighthouse, some variants are not a high enough contrast.

![image](https://user-images.githubusercontent.com/263432/89719798-265c5e00-d99a-11ea-8590-c9fbc3133787.png)

I am using a standard:

```jsx
<Button minWidth="sm" variantColor="blue">Text</Button>
```

with css reset and default theme. It appears a number of variant colors don't pass contrast ratios for lighthouse.
I did a little investigation and a few tools all confirmed that the button I had created with the default theme was not considered accessible. Here are a few testers that failed it:

https://color.a11y.com/Contrast/

https://www.aditus.io/button-contrast-checker/

The second tester also called out the contrast of the focus outline for blue as being too low.

I only tested with blue but when i passed the chakra docs through these testers it looked like other colors might also fail.

**Expected Behavior** Variants in the default theme to pass accessibility standards.

**Link to minimal reproduction** Please provide a codesandbox link or GitHub
repo with a minimal reproduction of the issue.

**Steps to Reproduce** Create a button using variantColor and run lighthouse on the page. Deploy the page to vercel and run either of the contrast checkers on it.

**Suggested solution(s)**

Bold text has less stringent contrast requirements but it seems like the solution is to darken the buttons with light text or lighten the buttons with black text.

**Comment by segunadebayo** [comment]
Thanks for reporting this @DavidJFelix 

@with-heart, let's plan to do a sweep across all components to check their color contrasts and improve them.

**Comment by with-heart** [comment]
@DavidJFelix Thanks for reporting! @segunadebayo let's do it!

**Comment by jmiazga** [comment]
I created a quick app to run the contrast tools against: https://chakra-buttons-a11y.vercel.app/

You can see the results here: https://www.aditus.io/button-contrast-checker/chakra-buttons-a11y-vercel-app-2020-08-10-at-20-15-46-345

**Comment by DavidJFelix** [comment]
This is great @jmiazga. You can see some of the background/text contrast I was mentioning -- I feel like some of these might be negated by the button font being "bold". A11y AA standards say 4.5:1 for text and 3:1 for "large" text which some interpretations have as bold text, others have as font size. AAA standards say 7:1 for text and 4.5:1 for large text. It seems like 4.5:1 is the goal contrast for lighthouse and aditus, so that might be a good first start.

I feel like 7:1 is probably borderline "high contrast mode". That being said, there is a darkmode/lightmode built in to chakra, high contrast could easily piggyback on themes.

**Comment by with-heart** [comment]
We added color exceptions for teal and yellow buttons in #1834. We looked at changing the default color for buttons to `.600` and scaling the other states from there, but weren't happy with the results.

A quick review of some other major sites in our space (like Tailwind) found that buttons were also considered inaccessible there.

I think this is something we need more community discussion on to decide whether we want to change our color scales, or make other changes to improve the accessibility score. For now, we're going to leave them where they are. I want to leave it open though as we should revisit this at a later point.

**Comment by segunadebayo** [comment]
Given the updates we've made to the button contrast, I think we can maintain this for now and look into it in the future.
